src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -181,27 +181,32 @@ def get_paths_to_platform_and_hwsku_dirs():
 
     return (platform_path, hwsku_path)
 
-
-def get_path_to_port_config_file():
+def get_path_to_port_config_file(asic=None):
     """
     Retrieves the path to the device's port configuration file
 
     Returns:
         A string containing the path the the device's port configuration file
     """
-    # Get platform and hwsku path
     (platform_path, hwsku_path) = get_paths_to_platform_and_hwsku_dirs()
 
-    # First check for the presence of the new 'platform.json' file
-    port_config_file_path = os.path.join(platform_path, PLATFORM_JSON_FILE)
-    if not os.path.isfile(port_config_file_path):
-        # platform.json doesn't exist. Try loading the legacy 'port_config.ini' file
-        port_config_file_path = os.path.join(hwsku_path, PORT_CONFIG_FILE)
-        if not os.path.isfile(port_config_file_path):
-            raise OSError("Failed to detect port config file: {}".format(port_config_file_path))
+    port_config_candidates = []
+
+    # Check for 'platform.json' file presence first
+    port_config_candidates_json.append(os.path.join(platform_path, PLATFORM_JSON))
```

Comment:
in the sonic-config-engine, [portconfig.py:71](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/portconfig.py#L71) there are two Json file one from docker root and another from platform root. Why are we not maintaining this behavior?

Reply:
Docker root is the same directory as platform root, just mounted to a different path inside the container. `get_paths_to_platform_and_hwsku_dirs()` figures out which path is available.